File: 백준/Bronze/14470. 전자레인지/전자레인지.js
const fs = require('fs');
const filePath = process.platform === 'linux' ? 'dev/stdin' : 'run/input.txt';
const input = fs.readFileSync(filePath).toString().trim().split('\n');

let [a, b, c, d, e] = input.map(Number);
let ice = Boolean(a < 0);
let r = 0;

while (a !== b) {
  if (a < 0) {
    a += 1;
    r += c;
  } else if (a === 0 && ice) {
    r += d;
    ice = false;
  } else {
    a += 1;
    r += e;
  }
}

console.log(r);

File: 백준/Bronze/14470. 전자레인지/전자레인지.py
a = int(input())
b = int(input())
c = int(input())
d = int(input())
e = int(input())

r = 0
ice = bool(a < 0)

while a != b:
    if a < 0:
        a += 1
        r += c
    elif a == 0 and ice:
        r += d
        ice = False
    else:
        r += e
        a += 1
else:
    print(r)